test/s3.js

```diff
@@ -5,7 +5,12 @@ var s3 = require('../src/js/s3');
 var fs = require('fs');
 var debug = require('gulp-debug');
 var through2 = require('through2').obj;
-var AWS = require('../node_modules/gulp-awspublish/node_modules/aws-sdk');
+var awspath = '../node_modules/gulp-awspublish/node_modules/aws-sdk';
+if (fs.existsSync(__dirname+'/../node_modules/aws-sdk/package.json')) {
+  awspath = '../node_modules/aws-sdk';
+}
+var AWS = require(awspath);
```

Comment:
If we depended on it directly, then I could just do `require('aws-sdk')` :D . Probably the cleanest solution would be to do just that and add `aws-sdk` as a *devDependency*. Actually, with NPM3 I could probably still do  `require('aws-sdk')` even without depending on it directly. That's a bit insane.